tdms/include/cell_coordinate.h

```diff
@@ -1,40 +1,31 @@
 /**
  * @file cell_coordinate.h
  * @author William Graham (ccaegra@ucl.ac.uk)
- * @brief Class declaration for Yee cell coordinates
+ * @brief Container for {ijk} or {IJK} grouped variables
  */
 #pragma once
 
-class CellCoordinate {
+#include <spdlog/spdlog.h>
 
-private:
-  // the indices of the Yee cell. Private so that they cannot be edited unintentionally
-  int cell_i = 0, cell_j = 0, cell_k = 0;
+/**
+ * @brief A structure for holding three values, which typically pertain to the same quantity but for each of the axial directions.
+ *
+ * Effectively stores the 3-vector (i,j,k). This is typically used to represent Yee cell indices, or 3D-array dimensions, or the maximum number of Yee cells in each coordinate direction, for example.
+ */
+struct ijk {
+  // The values (by dimension) that the object contains
+  int i = 0, j = 0, k = 0;
+
+  /** @brief Return the maximum of i,j,k */
+  int max() const { return std::max(std::max(i, j), k); }
 
-public:
-  /**
-   * @brief Construct a new Cell Coordinate object
-   *
-   * @param i,j,k The cell index to assign
-   */
-  CellCoordinate(int i = 0, int j = 0, int k = 0) {
-    cell_i = i;
-    cell_j = j;
-    cell_k = k;
-  };
-  /**
-   * @brief Construct a new Cell Coordinate object, passing in a contiguous block of memory
-   *
-   * @param indices The indices of the cell index to assign, in the order i,j,k
-   * @param buffer_start First index to read the indices from
-   */
-  CellCoordinate(int *indices, int buffer_start = 0) {
-    cell_i = indices[buffer_start];
-    cell_j = indices[buffer_start + 1];
-    cell_k = indices[buffer_start + 2];
-  };
+  /** @brief Print the (i,j,k) values */
+  void print() const { spdlog::info("ijk: ({},{},{})", i, j, k); }
 
-  int i() { return cell_i; };   //< fetch i
-  int j() { return cell_j; };   //< fetch j
-  int k() { return cell_k; };   //< fetch k
+  ijk &operator+=(int n) { this->i += n; this->j += n; this->k += n; return *this; }
```

Comment:
Cool. Do you... use this?

Reply:
[Here](https://github.com/UCL/TDMS/blob/d0e649e19c2fe043796f5a848b3a3dabe5cf338e/tdms/tests/unit/field_tests/test_Field_interpolation.cpp#L67) and [here](https://github.com/UCL/TDMS/blob/d0e649e19c2fe043796f5a848b3a3dabe5cf338e/tdms/tests/unit/field_tests/test_Field_interpolation.cpp#L280). There's also a lot of scope to use this in `execute()` and/or the setup of the variables when we get around to the nitty-gritty refactoring.